### PR TITLE
Adds telemetry functionality

### PR DESCRIPTION
**Pull request by josh-richardson**
Sends telemetry for all command types used in Web3j, stringifies secondary params. Probably not quite ready for merge yet. Needs a bit of refactoring, cleanup & review, cross-platform testing (particularly on windows due to spawning a java process, we may need to use `javaw` instead). Associated server-side code is already in place.

### CODE REVIEW

File: src/main/java/org/web3j/console/Runner.java [hunk 1]
```diff
@@ -48,7 +51,14 @@
     public static void main(String[] args) throws Exception {
         System.out.println(LOGO);
 
-        CliConfig config = CliConfig.getConfig(CliConfig.getWeb3jConfigPath().toFile());
+        CliConfig config = CliConfig.getConfig(CliConfig.getConfigPath().toFile());
+
+        if (Arrays.asList(args).contains("--telemetry")) {
+            new Telemetry(config, args).uploadAnalytics();
+        } else {
+            Telemetry.invokeAnalyticsUpload(args);
```
Comment (cfelde) [inline src/main/java/org/web3j/console/Runner.java]: What's the deal here? So someone passes in --telemetry and something is uploaded, but else something is also uploaded?

Comment (josh-richardson) [reply src/main/java/org/web3j/console/Runner.java]: I should refactor this. Basically invokeAnalyticsUpload spawns a background Web3j process which is invoked with `--telemetry`. Then the child process will see that it's been invoked with `--telemetry`, and perform the upload.

Comment (cfelde) [reply src/main/java/org/web3j/console/Runner.java]: Ah, right, I get it now

File: build.gradle [hunk 2]
```diff
@@ -86,3 +86,18 @@ dependencies {
...
+}
+
+shadowJar {
+    dependsOn shadowBugWorkaround
+    zip64 = true
+    // Shadow plugin bug: explodes the nested jar.
+    // Workaround: double-nest the jar.
+    // Refer https://github.com/johnrengelman/shadow/issues/111
+    from shadowBugWorkaround
+}
```
Comment (josh-richardson) [inline build.gradle]: As we have a `gradle-wrapper.jar` in the resources which is used during project generation, Shadow explodes this and causes a NPE by default, this workaround causes the jar file to be embedded normally as a resource.

**Comment by josh-richardson** [comment]
Superseded by epirus-cli

**Pull request closed without merging**